build_runner/test/daemon/daemon_test.dart

```diff
@@ -172,8 +176,8 @@ main() {
         ..registerBuildTarget(webTarget)
         ..startBuild();
       clients.add(client);
-      var buildResults = await client.buildResults.first;
-      expect(buildResults.results.first.status, BuildStatus.started);
+      await client.buildResults
+          .firstWhere((b) => b.results.first.status == BuildStatus.succeeded);
```

Comment:
We should probably use `expectLater` and `emitsThrough` here instead of this to get better failure messages (same below)